src/gui/ScreenHandler.cpp

```diff
@@ -164,6 +164,20 @@ void Screens::PushBeforeCurrent(const ScreenFactory::Creator screen_creator) {
 }
 
 void Screens::Loop() {
+    if (GuiDefaults::menu_timeout_enabled) {
+        window_frame_t *w = current.get();
+        if (w->IsTimeout()) {
+            gui_timeout_id = gui_get_menu_timeout_id();
+            if (gui_timer_expired(gui_timeout_id) == 1) {
+                close_all = true;
```

Comment:
You cannot close all screens just because last screen can be closed on menu timeout.